pkg/kv/bulk/bulkpb/ingestion_performance_stats.go

```diff
@@ -40,6 +50,38 @@ func (s *IngestionPerformanceStats) Identity() bulk.TracingAggregatorEvent {
 	return &stats
 }
 
+// getCombinedHist returns a new HistogramData that contains the currentHist
+// combined with the recordValue. If currentHist is nil, a new histogram is
+// initialized.
+func getCombinedHist(
+	currentHist *HistogramData, recordValue int64, dataType HistogramDataType,
+) *HistogramData {
+	var hist *hdrhistogram.Histogram
+	if currentHist != nil {
+		hist = hdrhistogram.Import(&hdrhistogram.Snapshot{
+			LowestTrackableValue:  currentHist.LowestTrackableValue,
+			HighestTrackableValue: currentHist.HighestTrackableValue,
+			SignificantFigures:    currentHist.SignificantFigures,
+			Counts:                currentHist.Counts,
+		})
+	} else if dataType == HistogramDataTypeLatency {
+		hist = hdrhistogram.New(minLatency.Nanoseconds(),
+			maxLatency.Nanoseconds(), sigFigs)
+	} else if dataType == HistogramDataTypeBytes {
+		hist = hdrhistogram.New(minBytes, maxBytes, sigFigs)
+	}
+	_ = hist.RecordValue(recordValue)
+	// Return the snapshot of this new merged histogram.
+	cumulativeSnapshot := hist.Export()
```

Comment:
One thing we discussed is a possible optimization which would be to avoid calling Export() on every call here.  Rather, we would have a private field that would store the hdrHistorgram state.  Then we would have a custom marshsaller  (or some method you call before marshal) that would take would do the export.

That would save us from importing and exporting this histogram on every single call.  This is something for us to consider if we ever see this code showing up in CPU profiles or the like.